Dockerfile

```diff
@@ -6,27 +6,58 @@ ENV POSTGIS_MAJOR=2.3 \
 
 RUN apt-get -y update \
  && apt-get -y --no-install-recommends install \
+        autotools-dev \
+        automake \
+        autoconf \
         curl \
-        build-essential cmake \
+        git \
+        libtool \
+        libcunit1-dev \
+        xsltproc \
+        docbook-xsl \
+        docbook-mathml \
+        dblatex \
+        build-essential \
+        cmake \
+        ca-certificates \
+        unzip \
         # PostGIS build dependencies
-		postgresql-server-dev-$PG_MAJOR libxml2-dev libjson0-dev libproj-dev libgdal-dev \
+                postgresql-server-dev-$PG_MAJOR libxml2-dev libjson0-dev libproj-dev libgdal-dev \
  && rm -rf /var/lib/apt/lists/*
 
-WORKDIR /opt/geos
-RUN curl -o /opt/geos.tar.bz2 http://download.osgeo.org/geos/geos-$GEOS_VERSION.tar.bz2 \
- && tar xf /opt/geos.tar.bz2 -C /opt/geos/ --strip-components=1 \
+RUN cd /opt/ \
+ && curl -o /opt/geos.tar.bz2 http://download.osgeo.org/geos/geos-$GEOS_VERSION.tar.bz2 \
+ && mkdir /opt/geos \
+ && tar xf /opt/geos.tar.bz2 -C /opt/geos --strip-components=1 \
+ && cd /opt/geos/ \
  && ./configure \
  && make -j \
  && make install \
  && rm -rf /opt/geos*
 
-WORKDIR /opt/postgis
-RUN curl -o /opt/postgis.tar.gz http://download.osgeo.org/postgis/source/postgis-$POSTGIS_VERSION.tar.gz \
- && tar xf /opt/postgis.tar.gz -C /opt/postgis --strip-components=1 \
- && ./configure --with-raster --with-topology \
+RUN cd /opt/ \
+ && curl -L https://github.com/google/protobuf/archive/v3.0.2.tar.gz | tar xvz && cd protobuf-3.0.2 \
+ && ./autogen.sh \
+ && ./configure \
+ && make \
+ && make install \
+ && ldconfig
+
+RUN cd /opt/ \
+ && curl -L https://github.com/protobuf-c/protobuf-c/releases/download/v1.2.1/protobuf-c-1.2.1.tar.gz | tar xvz && cd protobuf-c-1.2.1 \
+ && ./configure \
  && make \
  && make install \
  && ldconfig
 
+RUN cd /opt/ \
+ && git clone -b asmvt https://git.osgeo.org/gogs/bjornharrtell/postgis.git \
```

Comment:
As I see this[1]  branch is equal with [Postgis:svn-trunk](https://github.com/postgis/postgis/commits/svn-trunk) **~"Commits on Feb 8, 2017",** but the later MVT fixes not included. 

Imho it would be better to use  the current  [Postgis:svn-trunk](https://github.com/postgis/postgis/commits/svn-trunk) as a `POSTGIS_MAJOR=2.4dev` version.

[1]https://git.osgeo.org/gogs/bjornharrtell/postgis/commits/asmvt